drivers/fastboot/fastboot.c

```diff
@@ -0,0 +1,558 @@
+/*
```

Comment:
As stated in the HiKey pull request, it should not be necessary to execute Fastboot in the secure world so I'm not convinced this driver needs to exist in the TF upstream. This could be part of a richer NS_BL2U that uses existing normal world firmware functionality that integrates Fastboot (e.g. UEFI or UBoot).